File: src/_02_StructuralPattern/_02_06_FlyWeight/java/after/Character.java
package _02_StructuralPattern._02_06_FlyWeight.java.after;

public class Character {
  
  private char value;
  private String color;
  private Font font;

  public Character(char value, String color, Font font) {
    this.value = value;
    this.color = color;
    this.font = font;
  }

  public char getValue() {
    return value;
  }

  public String getColor() {
    return color;
  }

  public Font getFont() {
    return font;
  }

  @Override
  public String toString() {
    return "문자: " + value
      + ", 색상: " + color
      + ", 폰트: " + font.getFontFamily()
      + ", 사이즈: " + font.getFontSize();
  }
  
}

File: src/_02_StructuralPattern/_02_06_FlyWeight/Character.js
/**
 * @typedef { import("./Font").default } Font
 */

export default class Character {
  /** @type { string } */
  #value;

  /** @type { string } */
  #color;

  /** @type { Font } */
  #font;

  /**
   * @param { string } value 
   * @param { string } color 
   * @param { Font } font 
   */
  constructor(value, color, font) {
    this.#value = value;
    this.#color = color;
    this.#font = font;
  }

  /** @returns { string } */
  getValue() {
    return this.#value;
  }

  /** @returns { string } */
  getColor() {
    return this.#color;
  }

  /** @returns { Font } */
  getFont() {
    return this.#font;
  }

  /** @returns { string } */
  toString() {
    return `문자: ${this.#value}`
      + ` // 색: ${this.#color}`
      + ` // 폰트: ${this.#font.getFontFamily()}`
      + ` // 사이즈: ${this.#font.getFontSize()}`;
  }
}